pyanaconda/modules/subscription/satellite.py

```diff
@@ -0,0 +1,179 @@
+    are usually required in rhsm.conf to connect to a customer run Satellite instance
+    instead of to Hosted Candlepin for subscription purposes.
+
+    Each Satellite instance thus hosts a provisioning script available over plain
+    HTTP that client machines can download and execute. This script has minimal dependencies
+    and provisions the machine to be able to talk to the one given Satellite instance
+    by installing it's self signed certificates and adjusting rhsm.conf.
+
+    NOTE: As the script is downloaded over plain HTTP it is advised to ever only
+          provision machines from a Satellite instance on a trusted network, to
+          avoid the possibility of the provisioning script being tempered with
+          during transit.
+
+    :param str satellite_url: Satellite instance URL
+    :param proxy_url: proxy URL to use when fetching the script
+    :type proxy_url: str or None if not set
+    :returns: True on success, False otherwise
+    """
+    # make sure the URL starts with protocol
+    if not satellite_url.startswith("http"):
+        satellite_url = "http://" + satellite_url
+
+    # construct the URL pointing to the provisioning script
+    script_url = satellite_url + PROVISIONING_SCRIPT_SUB_PATH
+
+    log.debug("subscription: fetching Satellite provisioning script from: %s", script_url)
+
+    headers = {"user-agent": constants.USER_AGENT}
+    proxies = {}
+    provisioning_script = ""
+
+    # process proxy URL (if any)
+    if proxy_url is not None:
+        try:
+            proxy = ProxyString(proxy_url)
+            proxies = {"http": proxy.url,
+                       "https": proxy.url}
+        except ProxyStringError as e:
+            log.info("subscription: failed to parse proxy when fetching Satellite"
+                     " provisioning script %s: %s",
+                     proxy_url, e)
+
+    with util.requests_session() as session:
+        try:
+            # NOTE: we explicitly don't verify SSL certificates while
+            #       downloading the provisioning script as the Satellite
+            #       instance will most likely have it's own self signed certs that
+            #       will only be trusted once the provisioning script runs
+            result = session.get(script_url, headers=headers,
+                                 proxies=proxies, verify=False,
+                                 timeout=constants.NETWORK_CONNECTION_TIMEOUT)
+            if result.ok:
+                provisioning_script = result.text
+                result.close()
+                log.debug("subscription: Satellite provisioning script downloaded (%d characters)",
+                          len(provisioning_script))
+                return provisioning_script
+            else:
+                log.debug("subscription: server returned %i code when downloading"
+                          " Satellite provisioning script", result.status_code)
+                result.close()
```

Comment:
I think the context manager will handle that, so you don't need to explicitly close the result. See: https://2.python-requests.org/en/master/api/#requests.Response.close

Reply:
Sounds good, lets drop that. :)